### PR TITLE
Create python notebook to explore processed NOM data

### PR DESCRIPTION
**Pull request by samobermiller**
### All Submissions:

* [ Y] Have you followed the guidelines in our [Contributing document](CONTRIBUTING.md)?
* [Y] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/microbiomedata/nmdc_notebooks/pulls) for the same update/change?
* [Y] Does your PR link to an issue?
* [ Y] Have you described the changes this PR will make?

### New Notebook Submissions:

* [ Y] Have you included a summary of the notebook in the README.md included updated links to the notebook?
* [Y ] Does your PR include links to the new notebook **(in the branch)** for review using [nbviewer](https://nbviewer.jupyter.org/), [Colab](https://colab.research.google.com/), and [reviewnb](https://www.reviewnb.com/)? These three are the preferred ways to review changes and additions to notebooks during review.

### CODE REVIEW

**Review by kheal — CHANGES_REQUESTED** [review]

**Comment by samobermiller** [comment]
<p>I thought the random state setting was the seed?</p>

---

 <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/#comment-2341938319'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by kheal** [comment]
<p>Aha!  You're right :)</p>

---

 <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/#comment-2341938319'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: NOM_visualizations/README.md
[the remark this answers is not in the document]
Comment (samobermiller) [reply NOM_visualizations/README.md:1]: So should I just leave the links blank for now?

File: NOM_visualizations/python/nom_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,3353 @@
+{
```
Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <p>Wow great. I didn't realize you split these out. I am wondering if we should split them out even more so that the taxonomic python notebook can import them too. I think that's work for another time though. Just writing my thought here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <ol><li>Maybe clarify <code>NMDC metadata endpoint</code> to <code>NMDC metadata collection_name endpoint</code> or something similar, since there are lots of metadata endpoints?</li><li>Also, I don't see a file called <code>nom_functions.py</code> in this folder? Maybe I am missing it. Or is it possible you mean the <code>nmdc_api.py</code> file?</li><li>I'd also recommend putting the get_next_results with tic marks since its a function name: <code>get_next_results</code>.</li><li>When you say "we can filter <code>data_object_set</code> by type "FT ICR-MS Analysis Results". I would rephrase because we have a <code>type</code> slot in the <code>DataObjectSet</code> which is not what you are using to filter, so this might be confusing. I would write the full slot name: "we can filter <code>data_object_set</code> by <code>data_object_type</code> using keyword search of "FT ICR-MS Analysis Results".</li><li>Can you write a short blurb about why you are grabbing the <code>id</code>,   <code>md5_checksum</code> and <code>url</code> fields? </li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <p>Also, maybe explain why are grabbing the <code>has_input</code>.Its nice to set the reader up for understanding your thought process behind traversing the schema. E.g. The <code>has_input</code> will be important for grabbing the raw data objects that are the input into the Nom Analysis.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <p>Again, consider mentioning why you are grabbing the <code>has_input</code>  as well.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <p>I left a comment on Bea's notebook too, so maybe you guys should discuss. Your code assumes that the input into the <code>omics_processing_set</code> will be a Biosample, but the schema also allows for <code>ProcessedSample</code>s to be an input. This can get really messy (look at the taxonomy notebook). But because you aren't checking for processed samples, you may be missing data. I think we can leave this alone for this notebook. I believe the only study that takes advantage of the ProcessedSamples is NEON. So if there are NEON samples that went through NOM, your notebook would be missing that data. We can discuss in tomorrow's meeting, but hopefully there aren't NEON samples, or maybe we can just ignore this for now since we already have a notebook that shows that really crazy schema traversal.</p><p> </p><p>One thing we could do which Bea mentioned is to use this endpoint: <a href="https://api.microbiomedata.org/docs#/metadata/get_collection_name_by_doc_id_nmdcschema_ids__doc_id__collection_name_get" target="_blank">https://api.microbiomedata.org/docs#/metadata/get_collection_name_by_doc_id_nmdcschema_ids__doc_id__collection_name_get</a> to get the collection names that the <code>omice_processing_set</code> <code>has_input</code> belongs to. If there are not <code>ProcessedSamples</code>. and just <code>Biosamples</code>. Then we're golden.</p><p> </p><p>Maybe we can also assume there are no ProcessedSamples because your code didn't break in the next API call?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <p>I like the way you did this and that it is different from the other notebooks.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <p>Can you explain the axes a little more? Count vs. Assigned Peak Count.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (brynnz22) [inline NOM_visualizations/python/nom_data.ipynb:1]: <p>Can you just explain your thinking behind this a bit. Why did you filter to datasets with at least 250 identified peaks?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microbiomedata/nmdc_notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

**Comment by kheal** [comment]
@samobermiller  Please add your new notebook to the python gh_workflow as part of this PR :).  Add your notebook to the bottom of this file to do so. https://github.com/microbiomedata/nmdc_notebooks/blob/main/.github/workflows/notebook_check_python.yml

File: NOM_visualizations/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Exploring NOM Metadata and Visualization via the NMDC Runtime API
+
+This folder includes two notebooks (in R and Python) that demonstrate how metadata from natural organic matter (NOM) can be gathered via the [NMDC-runtime API](https://api.microbiomedata.org/docs) and analyzed.
+
+## Python
+- [Static rendered Jupyter notebook](https://nbviewer.org/github/microbiomedata/nmdc_notebooks/blob/main/NOM_visualizations/python/nom_data.ipynb). _Viewing only, not editable_
+- [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/microbiomedata/nmdc_notebooks/blob/main/NOM_visualizations/python/nom_data.ipynb).This is the recommended way to interact with the notebook. _You need a google account to use this option_
```
Comment (kheal) [inline NOM_visualizations/README.md:7]: You might want to add a note that although google colab does work, it will be slow.  And point people to the other option.  You can see the taxonomy readme for that language.  You can test out the google colab here: https://colab.research.google.com/github/microbiomedata/nmdc_notebooks/blob/33-create-python-notebook-to-explore-processed-data-for-a-second-omics-data-type/NOM_visualizations/python/nom_data.ipynb#scrollTo=TN34Q-BwMAg1

**Review by kheal — COMMENTED** [review]
Notebook itself looks great!

Checking over the other files.

A couple small changes
1) You need to add your new requirements to both the requirement-dev and requirements.  If you go to https://colab.research.google.com/github/microbiomedata/nmdc_notebooks/blob/33-create-python-notebook-to-explore-processed-data-for-a-second-omics-data-type/NOM_visualizations/python/nom_data.ipynb#scrollTo=TN34Q-BwMAg1 (where the google colab link will take you), you can it won't work because it's missing fastcluster).
2) I'm still seeing the __pycache__ in this branch.  I think you added it to the gitignore after you pushed it (very common mistake), but we need to remove it from the repo.  To do that, delete it locally, remove it from the gitignore, commit, push.  _Then_ add it to your .gitignore, re-render, commit, push.

**Review by kheal — APPROVED** [review]
